### ISSUE TITLE
'Picture Frames' Animated Titles error

### ISSUE THREAD

**Issue by badukibaduki** [OpenShot/openshot-qt#3420]
**Describe the bug**
'Picture Frames' from the Animated Titles doesn't work.
I have tried with different versions of Openshot(2.5.1 & 2.5.1-dev2) and Blender(2.82 & 2.90alpha) but same outcome. Both programs run fine except occasional Openshot shutdowns but that's fine.
All 'Animated Titles' also work fine. So why is this 'Picture Frames' not working? There are no examples of doing the picture frames and so I can't even see if I'm doing anything wrong. I have also tried this with one other friend and he too have the same problem. Unfortunately he has more problems then I with frequent shutdowns and longer rendering, output time even though he has a better cpu. But that's not the problem right now. I need to get this picture frames working. 

**Steps to reproduce the behavior:**
1. Go to 'Title'
2. Click on 'Animated Titles'
3. Scroll down to 'Picture Frames'
4-1. No images in the project produces blank 4 image slots with no files found path (unclickable) but render-able with pink blocks.
4-2. If images are loaded then there is nothing in the paths section and render button is grayed out. (see screenshots)

**Expected behavior**
I have tried with different versions of Openshot(2.5.1 & 2.5.1-dev2) and Blender(2.82 & 2.90alpha) but same outcome. 
Both programs run fine except occasional Openshot shutdowns. 
All 'Animated Titles' are also fine except 'Picture Frames'

**System Details**
- OpenShot Version: 2.5.1-dev2
- libopenshot Version:  0.2.5-dev2
- Platform: Windows-10-10.0.18362
- Processor: Intel64 Family 6 Model 60 Stepping 3, GenuineIntel
- Machine: MSI i7
- Python version: 3.7.4
- Qt5 version: 5.12.4
- PyQt5 version: 5.12.3
- Qt Detected Languages: ['ko-KR', 'en-US']
- LANG Environment Variable: 
- LOCALE Environment Variable: 
- Daily Build Installed: http://github.com/OpenShot/openshot-qt/releases/download/OpenShot-v2.5.1-dev2-1588081875-15c3efb7-26b7db94-x86_64.exe

**Log Files**
- http://logs.openshot.org/2o9nripnebx0l5p9llb1topueymyn94j/openshot-qt.zip
- http://logs.openshot.org/2o9nripnebx0l5p9llb1topueymyn94j/libopenshot.zip

**Exception / Stacktrace**
```No stacktrace found in log files```

**Screenshots** 
No picture loaded:
![image](https://user-images.githubusercontent.com/64521083/80569628-4ba10e00-8a34-11ea-864e-98a8d9538c82.png)

Picture loaded:
![image](https://user-images.githubusercontent.com/64521083/80569571-2b714f00-8a34-11ea-848f-a9308b88b6e3.png)

**Comment by ferdnyc** [comment]
@badukibaduki From the log you posted, that's almost certainly my fault, sorry. I'll take a look right now and hopefully have it fixed shortly.

**Comment by ferdnyc** [comment]
#3421 should fix both of the errors in your log, I should have an updated Windows installer for you in just a few minutes.

**Comment by ferdnyc** [comment]
Here's a link to a Windows 64-bit installer with the fixes from #3421 included:

https://drive.google.com/file/d/1soB_rNptIVkpnpo-aO9Zp_xNNQt1M8Uz/view?usp=sharing

Let us know if that fixes the problem, I tested under Linux briefly and things seem to be working correctly now. Sorry about that!

**Comment by badukibaduki** [comment]
Oh wow! Thank you so much! It's working! 
But is this normal? I have 4 images but only one is pre-selected in the path? 
Though I can click them and choose. Just curios if that is also a bug....or just is.
But thank you for quick fix!!!!!!!

![image](https://user-images.githubusercontent.com/64521083/80601689-17941000-8a69-11ea-98f3-45ebf97ab7b9.png)

**Comment by ferdnyc** [comment]
@badukibaduki Great, thanks for getting back to me on that!

Yeah, the image selection thing is normal, we don't assume which files you'd want to choose for the titles or in which order you'd want to use them, so all of the selectors just default to the first item on the list.

I suppose we could be a _little_ proactive, and automatically choose the first four items in order. At worst you'd have to adjust them all, which would be no different from how things currently work.

I'll have to think about that a bit. It should be possible, as long as I can't think of any problems we'd be causing for people with that change.

**Comment by badukibaduki** [comment]
Yes true, but when I first tried to use the 'Picture Frames', I didn't load any images and thought I could click at the file path and choose them. (since the name 'Picture 1 Path' suggests I can choose any image in any folders) But if we are using images from the loaded project files, it should be just 'Picture 1' 'Picture 2' and so on.. without 'path' which confuses a little or if you could implement in choosing any file. It would look like this.. just one more option for you to think about? ;)
![openshot](https://user-images.githubusercontent.com/64521083/80620975-a1e86e00-8a81-11ea-9fff-48e341368ed7.png)

**Comment by ferdnyc** [comment]
That's an excellent point, and I will definitely at least remove "Path" from the labels. (I suspect originally you _could_ load image files directly, and when the interface was changed to work off of the Project Files list, the labels were never corrected

In theory, there's no reason the images _need_ to be in Project Files. Once the animation is generated, they're no longer strictly necessary. However, I think managing them through Project Files is a good idea for two reasons:
1. All of the items in Project Files have been scanned as valid media files, so we don't have to worry about the user loading something incompatible. Which if it's something like a ZIP, DLL, or EXE, could even potentially crash Blender. So even with a direct "load" option, we'd still want to route new files through Project Files so they can be validated. But updating the Project Files list while the Animated Titles dialog is open isn't really supported, currently.
1. Using items from the Project Files list ensures that any images used are considered project media. If the files were loaded directly and only used to generate the title animation, once the window closed there'd be no real record of which files are used in that animation. Loading them from the Project Files list at least means they're tracked somewhere. While the user could certainly remove them afterwards, hopefully they'll realize it's better not to, just in case they decide to re-do the title animation later on. (Or if they need to make a list of every content file used in the project for any reason — credit, archiving, etc.)

**Comment by ferdnyc** [comment]
Part of me is tempted to change the selection boxes to initially read "Choose Project File...", to make it a bit more explicit what's expected there. But then you'd be _forced_ to make a selection for all four pictures every time, and that goes against the OpenShot design guideline that _something_ useful should be produced even with all-default settings.

(As you've seen, when there are no project files loaded the lists are disabled and show "No Files Found". Which maybe could be changed to "No Project Files available" or something, but otherwise I think is clear enough.)

**Comment by ferdnyc** [comment]
I've updated PR #3421 to also fix those labels (to just "Picture 1", "Picture 2", etc.).

**Comment by badukibaduki** [comment]
Cool! I corrected my xml as yours for now instead of downloading the whole daily build. Yes I agree with you. But in my case, I was only using those 4 images for the title animation. 4 can be easily remembered and also redo it anytime without being saved. But having them in the project files bin confuses me with bunch of other media files. If I have 10+ images for slide show, I will have to know them by file names (since thumbnail view is not available in the list) and that will be very frustrating. For testing, I used 30 images and the whole screen took up with the list and I have to know which file name is which. I also tried selecting 4 images from the bin and open 'Animated titles' as I thought it would limit me in using only those 4 but that didn't do anything. I still had to choose from the long list. But, if I want to re-use the image for both the title and the slideshow, that would be fine. So, pre-selecting 4 for the title would be an option, loading directly is an option even though validation could cause openshot to crash but can limit to only view/select image files and no other visible. Or, if none will work... then maybe instead of "No files found".. how about.. "Add files to Project" something like that.
Anyhow, you guys are doing a great job!!!!
